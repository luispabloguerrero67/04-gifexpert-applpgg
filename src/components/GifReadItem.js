import React from 'react'
import PropTypes from 'prop-types'
//import ImageList from '@mui/material/ImageList';
import ImageListItem from '@mui/material/ImageListItem';
import ImageListItemBar from '@mui/material/ImageListItemBar';

const GifReadItem = ({ id, title, url }) => {
    return (
        <div className='card animate__animated animate__fadeIn'>

            <ImageListItem key={id}>
                <img
                    src={`${url}?w=248&fit=crop&auto=format`}
                    srcSet={`${url}?w=248&fit=crop&auto=format&dpr=2 2x`}
                    alt={title}
                    loading="lazy"
                />
                <ImageListItemBar
                    title={title}
                    
                    position="below"
                />
                <p>{title}</p>
            </ImageListItem>
        </div>
    )
}

GifReadItem.propTypes = {
    //id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

export default GifReadItem