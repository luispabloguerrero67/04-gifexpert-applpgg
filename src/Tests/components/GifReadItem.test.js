import { shallow } from "enzyme"
import { shallowToJson } from "enzyme-to-json"
import React from 'react';
import GifReadItem from '../../components/GifReadItem'

describe('Pruebas en <GifGridItem/>', () => {

    const title = 'title';
    const url = 'https://localhost/logo.jpg';
    const id = 1;
    const wrapper = shallow(<GifReadItem title={title} url={url} id={id} />);
    test('debe de mostrar el componente correctamente', () => {


        expect(wrapper).toMatchSnapshot();
    });

    test('debe de tener un parrafo con el title', () => {

        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );

    });

    test('debe tener imagen con url y alt de los props',() =>{

        const img = wrapper.find('img');

        expect(img.props('alt').alt).toBe(title);
    })

    test('debe de tener animate_fadeind', () => { 
        
        const div = wrapper.find('div');
        const props = div.props('Classname').className;

        expect(props.includes('animate__fadeIn')).toBe(true);

        console.log(props);
     })
});