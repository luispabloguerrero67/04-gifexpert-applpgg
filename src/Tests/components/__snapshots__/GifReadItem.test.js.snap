// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <ForwardRef(ImageListItem)
    key="1"
  >
    <img
      alt="title"
      loading="lazy"
      src="https://localhost/logo.jpg?w=248&fit=crop&auto=format"
      srcSet="https://localhost/logo.jpg?w=248&fit=crop&auto=format&dpr=2 2x"
    />
    <ForwardRef(ImageListItemBar)
      position="below"
      title="title"
    />
    <p>
      title
    </p>
  </ForwardRef(ImageListItem)>
</div>
`;
